runtime/oti/ObjectMonitor.hpp

```diff
@@ -279,10 +280,22 @@ class VM_ObjectMonitor
 		bool locked = false;
 		if (LN_HAS_LOCKWORD(currentThread, object)
 #if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
-		&& !J9_IS_J9CLASS_VALUETYPE(J9OBJECT_CLAZZ(currentThread, object))
+		&& !J9_IS_J9CLASS_VALUETYPE(J9OBJECT_CLAZZ(currentThread, object)
```

Comment:
this should just be `!J9_DOES_CLASS_DISALLOW_LOCKING(clazz)` no need to add the exception code to the fast path as if this fails it will call the slow path

Reply:
all the exception code can be put in the slow path

Reply:
Fixed.